enterprise-suite/tests/smoke_alert

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+#set -x
+
+## Test alerting
+
+NAMESPACE=${NAMESPACE:=lightbend}
+
+source smokecommon
+source prometheus_common
+
+# 1. Create new configmap for alertmanager deployment
+# 2. Cache existing configmap setting in deployment
+# 3. Update alertmanager deployment configs to use new configmap
+# 4. run tests
+# 5. Put original configmap back in place
+
+export alertmanager_smoke_config_dir=$(mktemp -d -t configmap.XXX)
+
+# As per https://developer.lightbend.com/docs/console/current/installation/alertmanager.html#2-create-the-configmap
+make_alertmanager_configmap() {
+  cp resources/alertmanager.yml $alertmanager_smoke_config_dir
+  cp ../alertmanager/*.tmpl $alertmanager_smoke_config_dir
+  kubectl -n $NAMESPACE create configmap alertmanager-smoke-config --from-file=$alertmanager_smoke_config_dir --dry-run -o yaml | kubectl apply -f -
+}
+
+make_alertmanager_configmap
+ORIGINAL_AM_CONFIGMAP=$( kubectl get deployment prometheus-alertmanager -n $NAMESPACE -o go-template --template="{{ (index .spec.template.spec.volumes 0).configMap.name }}" )
+kubectl patch deployment prometheus-alertmanager -n $NAMESPACE -p '{"spec":{"template":{"spec":{"volumes":[{"name":"config-volume", "configMap":{"name":"alertmanager-smoke-config"}}]}}}}'
+
+#
+# launch test apps
+#
+
+set -e
+for app in resources/alert-*.yaml ; do
+  kubectl apply -f "$app" -n $NAMESPACE --wait
+done
+
+# Wait until all deployments are up and running
+busy_wait deployments_up
+
+cleanup() {
+  for app in resources/alert-*.yaml ; do
+    kubectl delete -f "$app" -n $NAMESPACE
+  done
+  # Put alertmanager configmap back.
+  if [ -n "$ORIGINAL_AM_CONFIGMAP" ] ; then
+    kubectl patch deployment prometheus-alertmanager -n $NAMESPACE -p '{"spec":{"template":{"spec":{"volumes":[{"name":"config-volume", "configMap":{"name":"'$ORIGINAL_AM_CONFIGMAP'"}}]}}}}'
+  fi
+  if [ -n "$alertmanager_smoke_config_dir" ] ; then
+    rm -rf $alertmanager_smoke_config_dir
+  fi
+}
+
+trap cleanup 0
+set +e
+
+busy_wait prom_three_scrapes kube_pod_info
+T $? Prometheus kube-state-metrics has at least three scrapes
+
+PROM_SCRAPES=$( prom_scrapes )
+[[ PROM_SCRAPES -ge 4 ]]
+T $? "Prometheus scraping $PROM_SCRAPES targets"
+
+# app tests
+app_instances='count( count by (instance) (ohai{es_workload="es-alert-test", namespace="'$NAMESPACE'"}) ) == 2'
```

Comment:
does the 2 replicas thing here signify anything special in the alert tests? just thinking about pruning down resources, no biggie either way

Reply:
Nope.  Changed to one replica.